### PR TITLE
Pixel Projectiles [WIP]

### PR DESCRIPTION
**Pull request by mattroks101**
## About The Pull Request

Pixel projectiles system. Roughly ported from Aurora.

This adds a pixel projectile system like what TG and Aurora have. It functions the same but projectiles are more accurate to where you click, move a little faster, and they run off a subsystem reducing slowdown.

### CODE REVIEW

File: code/modules/organs/internal/xenos/acid_gland.dm [hunk 1]
```diff
@@ -43,7 +43,7 @@
 		return
 
 	var/obj/item/projectile/energy/neurotoxin/A = new /obj/item/projectile/energy/neurotoxin(usr.loc)
-	A.current = U
+	//A.current = U
```
Comment (mattroks101) [inline code/modules/organs/internal/xenos/acid_gland.dm]: Not sure why this broke. Or what this even does. Have to look into this.

File: cev_eris.dme [hunk 2]
```diff
@@ -2149,11 +2151,10 @@
 #include "code\modules\projectiles\_defines.dm"
 #include "code\modules\projectiles\ammunition.dm"
 #include "code\modules\projectiles\dynamic_cursor.dm"
-#include "code\modules\projectiles\effects.dm"
 #include "code\modules\projectiles\gun.dm"
 #include "code\modules\projectiles\gun_firemode.dm"
 #include "code\modules\projectiles\gun_hud_actions.dm"
-#include "code\modules\projectiles\projectile.dm"
+#include "code\modules\projectiles\projectile_new.dm"
```
Comment (mattroks101) [inline cev_eris.dme]: I left the original file there to help with debugging. It can be removed before this is merged.

**Comment by TorinoFermic** [comment]
Right now, there's a problem. It can generate a flood of runtimes if the targloc is nulled inside preparePixelProjectile().

**Comment by mattroks101** [comment]
I'll re-open this when I'm actually able to fix it.

**Comment by MadmanMartian** [comment]
that was close

**Comment by mattroks101** [comment]
I've still got it saved locally anyway. Maybe I'll dig it up this week and see if I can figure out the runtime issues.

**Pull request closed without merging**